packages/celer-web-app/src/app/AppDocumentProvider.tsx

```diff
@@ -58,17 +58,12 @@ export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({ servic
 					setStatus(null);
 					setRouteSourceBundle(doc);
 					// Store current page in localStorge if it is a GitHub route
-					if (params.user && params.repo) {
-						// Define the current url from the parameters
-						let url = "gh/" + params.user + "/" + params.repo;
-						if (params.branch) {
-							url += "/" + params.branch;
-						}
-						// Add this page to the recent pages list
+					const url = service.getDocPath();
+					if (url.startsWith("https://raw.githubusercontent.com/")) {
```

Comment:
service should be responsible for the replace logic here as well.
AppDocumentProvider should not care about service-specific logic